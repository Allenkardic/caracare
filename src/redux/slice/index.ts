export * from './charactersSlice';
export * from './favouriteCharactersSlice';
export * from './firstEpisodeSlice';
export * from './lastEpisodeSlice';
export * from './singleCharacterSlice';
export * from './settingsSlice';
