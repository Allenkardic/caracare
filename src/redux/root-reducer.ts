import {combineReducers} from '@reduxjs/toolkit';
import {charactersSliceReducer} from './slice';

const rootReducer = combineReducers({
  characters: charactersSliceReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
