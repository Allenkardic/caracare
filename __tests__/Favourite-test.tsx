import 'react-native';
import React from 'react';
import Favourite from '../src/screens/favourite/favourite';

import renderer from 'react-test-renderer';
import {testAppSelector} from '../src/redux/test-app-selector';
import {useAppSelector} from '../src/redux/redux-hooks';

jest.mock('../src/redux/redux-hooks');

const createTestProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe('Favourite Screen', () => {
  beforeEach(() => {
    useAppSelector.mockImplementation(testAppSelector);
  });
  afterAll(() => {
    jest.clearAllMocks();
  });
  it('renders correctly', () => {
    let props: any;
    props: createTestProps;
    const tree = renderer.create(<Favourite {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
