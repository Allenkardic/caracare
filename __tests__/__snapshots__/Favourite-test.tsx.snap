// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourite Screen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 24,
    }
  }
>
  <RCTScrollView
    CellRendererComponent={[Function]}
    ListEmptyComponent={
      <EmptyList
        subTitle="You do not have favourite character yet"
      />
    }
    ListHeaderComponent={
      <View
        style={
          Object {
            "height": 10,
          }
        }
      />
    }
    animatedStyle={
      Object {
        "value": Object {},
      }
    }
    collapsable={false}
    entering={
      FadeIn {
        "build": [Function],
        "durationV": 300,
        "randomizeDelay": false,
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
        tintColor="rgb(28, 28, 30)"
        title=""
        titleColor="rgb(28, 28, 30)"
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "rgb(242, 242, 242)",
        "flex": 1,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "height": 10,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginTop": 240,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#AEAEB6",
                "fontSize": 267,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï’µ
        </Text>
        <View
          style={
            Object {
              "marginTop": 48,
              "paddingHorizontal": 192,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 28,
                },
                Object {
                  "fontSize": 25,
                  "textAlign": "center",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            You do not have favourite character yet
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
